### PR TITLE
Update FxA documentation

### PR DESCRIPTION
**Pull request by clouserw**
This is the tip of the iceberg on updating our docs, but I wanted to pull out the obvious stuff since we have a few new people starting in a couple weeks.  There is still **plenty** of low hanging fruit here.

### CODE REVIEW

File: docs/product-portal/accounts/dev-process.md [hunk 1]
```diff
@@ -40,178 +40,87 @@ for the Application Services team:
...
-* [Active issues for Firefox Accounts](https://waffle.io/mozilla/fxa)
+We use labels and milestones to keep things organised.
 
 Issue status is reflected by the following:
 
-* The milestone indicates *why* we are working on this issue.
-* The waffle column indicates *what* the next action is and *when*
-  we expect to complete it.
+* The milestone indicates *when* we are working on this issue.
+* The issue itself will have updates indicating *what* the next action.
```
Comment (shane-tomlinson) [inline docs/product-portal/accounts/dev-process.md]: `action.` => `action is.`

File: docs/product-portal/accounts/dev-process.md [hunk 2]
```diff
@@ -40,178 +40,87 @@ for the Application Services team:
...
+* **skill:database**
+
+##### Blocked on another person/team/decision:
+* **needs:discussion** There is an open question blocking this issue from moving forward
+* **needs:legal**
+* **needs:ops**
+* **needs:product** Any product questions including missing content or copy
+* **needs:qa**
+* **needs:security**
+* **needs:visuals**  UI is in place but doesn’t look right
```
Comment (shane-tomlinson) [inline docs/product-portal/accounts/dev-process.md]: visuals seems pretty close in concept to ux. I've never actually used it.

File: docs/product-portal/accounts/dev-process.md [hunk 3]
```diff
@@ -271,56 +180,8 @@ Here are some handy questions and things to consider when reviewing code for Fir
...
-in a specific `train-nnn` branch,
-which must then be merged back to `master`.
-
-It's important that:
-
-1. The merge happens;
-
-2. It really is just a vanilla `git merge`
-   and not a `rebase`, `cherry-pick` or `merge --squash`.
+The release process is being [documented and reviewed](https://docs.google.com/document/d/1lAK-Wvmr7xPsVHIDVFhG5RlQKQQL1Z4NlhyCPLLq4-U/edit?ts=5cb8ba2d).  We'll update this official documentation when the details are sorted out.
```
Comment (shane-tomlinson) [inline docs/product-portal/accounts/dev-process.md]: Users have to unfortunately sign into GDocs to view this.

File: docs/product-portal/accounts/project-details.md [hunk 4]
```diff
@@ -39,43 +37,47 @@ Links:
 These fine folks are the globally distributed team at the core of Firefox Accounts
 development, and will be happy to help answer any questions you might have:
 
-* [Ryan Kelly](https://github.com/rfk) - Engineering (Melbourne, approx UTC+10)
 * [Shane Tomlinson](https://github.com/shane-tomlinson/) - Engineering (London, approx UTC)
-* [Vlad Filippov](https://github.com/vladikoff) - Engineering (Toronto, approx UTC-5)
 * [Vijay Budhram](https://github.com/vbudhram) - Engineering (Orlando, approx UTC-5)
 * [Phil Booth](https://github.com/philbooth) - Engineering (London, approx UTC)
+* [Danny Coates](https://github.com/dannycoates) - Engineering (West Coast US, approx UTC-8)
```
Comment (shane-tomlinson) [inline docs/product-portal/accounts/project-details.md]: What about you @clouserw, your name isn't in this list!

File: docs/product-portal/accounts/project-details.md [hunk 5]
```diff
@@ -39,43 +37,47 @@ Links:
...
 * [John Morrison](https://github.com/jrgm) - Operations (Mountain View, approx UTC-8)
 * [Jon Buckley](https://github.com/jbuck) - Operations (Toronto, approx UTC-5)
 * [Ryan Feeley](https://github.com/rfeeley) - UX (Toronto, approx UTC-5)
-* [Chris Karlof](https://github.com/ckarlof) - Identity Services Manager (San Francisco, approx UTC-8)
 * [Alex Davis](https://github.com/davismtl) - Product Manager (Mountain View, approx UTC-8)
```
Comment (shane-tomlinson) [inline docs/product-portal/accounts/project-details.md]: Should we add the folks on subscription services too?

File: docs/product-portal/accounts/project-details.md [hunk 6]
```diff
@@ -39,43 +37,47 @@ Links:
...
 We meet regularly to triage bugs and make grand plans for the future.  Anyone is welcome to
 join us in the following forums:
 
-* Regular video meetings, as noted on the [project calendar](https://www.google.com/calendar/embed?src=mozilla.com_urbkla6jvphpk1t8adi5c12kic%40group.calendar.google.com) and with minutes in the [coordination google-doc](https://docs.google.com/document/d/1r_qfb-D1Yt5KAT8IIVPvjaeliFORbQk-xFE_tRNM4Rc/)
+* Regular video meetings, as noted on the [project calendar](https://www.google.com/calendar/embed?src=mozilla.com_urbkla6jvphpk1t8adi5c12kic%40group.calendar.google.com)
 * The [Firefox Accounts mailing list](https://mail.mozilla.org/listinfo/dev-fxacct)
-* The `#fxa` channel on [Mozilla IRC](https://wiki.mozilla.org/IRC)
+* The `#fxa` channel on Slack or [Mozilla IRC](https://wiki.mozilla.org/IRC)
+
+Firefox Accounts has been around long enough that many people have contributed.
```
Comment (shane-tomlinson) [inline docs/product-portal/accounts/project-details.md]: This is a great section.

File: docs/product-portal/accounts/dev-process.md (hunk 2 above)
Comment (clouserw) [reply docs/product-portal/accounts/dev-process.md]: It was a request from John to start differentiating.  UX = flows, Visuals = images/icons on the page in a nutshell (I'm sure there is other nuance).

File: docs/product-portal/accounts/dev-process.md (hunk 3 above)
Comment (clouserw) [reply docs/product-portal/accounts/dev-process.md]: That's ok for now.  Once we finalize it we'll add it to the real docs

**Pull request merged**